File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache_arr = []
    if cacheSize == 0:
        return len(cities)*5
    for city in cities:
        city = city.upper()
        if city in cache_arr:
            answer+=1
            cache_arr.remove(city)
            cache_arr.append(city)
        elif len(cache_arr) < cacheSize:
            answer+=5
            cache_arr.append(city)
        else:
            answer+=5
            cache_arr.pop(0)
            cache_arr.append(city)
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    var answer = 0;
    let cache_arr = []
    if (cacheSize === 0) {
        return cities.length * 5
    }
    for (let city of cities) {
        city = city.toUpperCase()
        if (cache_arr.includes(city)) {
            answer += 1
            cache_arr = cache_arr.filter((e) => e !== city)
            cache_arr.push(city)
        } else if (cache_arr.length < cacheSize) {
            answer += 5
            cache_arr.push(city)
        } else {
            answer += 5
            cache_arr.shift()
            cache_arr.push(city)
        }
    }
    return answer;
}